python_ta/__init__.py

```diff
@@ -127,6 +129,114 @@ def check_all(
     )
 
 
+def _setup_linter(
+    local_config: Union[dict[str, Any], str],
+    load_default_config: bool,
+    output: Optional[str],
+) -> tuple[PyLinter, BaseReporter | MultiReporter]:
+    """Set up the linter and reporter for the check."""
+    linter = reset_linter(config=local_config, load_default_config=load_default_config)
+    current_reporter = linter.reporter
+    current_reporter.set_output(output)
+    messages_config_path = linter.config.messages_config_path
+    messages_config_default_path = linter._option_dicts["messages-config-path"]["default"]
+    use_pyta_error_messages = linter.config.use_pyta_error_messages
+    messages_config = load_messages_config(
+        messages_config_path, messages_config_default_path, use_pyta_error_messages
+    )
+
+    global PYLINT_PATCHED
+    if not PYLINT_PATCHED:
+        patch_all(
+            messages_config, linter.config.z3
+        )  # Monkeypatch pylint (override certain methods)
+        PYLINT_PATCHED = True
+    return linter, current_reporter
+
+
+def _check_file(
+    linter: PyLinter,
+    file_py: AnyStr,
+    local_config: Union[dict[str, Any], str],
+    load_default_config: bool,
+    autoformat: Optional[bool],
+    is_any_file_checked: bool,
+    current_reporter: BaseReporter | MultiReporter,
+    level: str,
+    f_paths: list,
+) -> tuple[bool, BaseReporter | MultiReporter, PyLinter]:
+    """Check the file that called this function."""
+    logging.basicConfig(format="[%(levelname)s] %(message)s", level=logging.INFO)
+    allowed_pylint = linter.config.allow_pylint_comments
+    if not _verify_pre_check(file_py, allowed_pylint):
+        return is_any_file_checked, current_reporter, linter
```

Comment:
Move this block out of the function. Its presence makes the rest of the code more complex.